docs/union.md

```diff
@@ -0,0 +1,150 @@
+# Unions
+
+Consider this Oomph code:
+
+    func foo(Str | null message):
+        if message == null:
+            print("No message")
+        else:
+            print(message.get())
+
+    export func main():
+        foo(null)       # prints "No message"
+        foo("hello")    # prints "hello"
+
+Let's go through this step by step.
+
+    func foo(Str | null message):
+
+Here `Str | null` is a union type. Read it as "`Str` or `null`".
+When the code runs, each instance of the union type has exactly one **active member**,
+such as `Str` when this function is called with the string `"hello"` as an argument.
+A union type can also contain multiple types, such as `Str | Int | Bool`.
+
+    if message == null:
+        print("No message")
+
+In this context, `null` has a different meaning than previously.
+Oomph has two different meanings for `null`:
+there's a variable named `null` which we are using here,
+and a type named `null` that we used previously.
+
+In this line, the left side of `==` has type `Str | null` and the right side has type `null`.
+This causes the `==` operator to [convert](implicit-conversions.md) `null` to type `Str | null`.
+
+Two objects with union types are equal when they have the same active member
+and the underlying values are equal.
+This ensures that `message == null` is a valid way to check
+whether `null` is the active member of `message`.
+
+    else:
+        print(message.get())
+
+Recall that `message` has type `Str | null`.
+However, when the `else` runs, the active member of `message` can't be `null`.
+The `.get()` method converts from `Str | null` to `Str`,
+causing a runtime error if the active member is `null`.
```

Comment:
How's that possible if you previously wrote that if message is null, the right side gets converted to Str/null and so the 'if' condition is met not the 'else' one?